app/models/generic_object_definition.rb

```diff
@@ -8,44 +8,81 @@ class GenericObjectDefinition < ApplicationRecord
     :time     => ActiveModel::Type::Time.new
   }.freeze
 
-  validates :name, :presence => true, :uniqueness => true
+  FEATURES = %w(attribute association).freeze
 
   serialize :properties, Hash
 
   has_one   :picture, :dependent => :destroy, :as => :resource
   has_many  :generic_objects
 
-  before_destroy :check_not_in_use
+  validates :name, :presence => true, :uniqueness => true
+  validate  :validate_properties
```

Comment:
I prefer to validate and normalize in `def properties=(props)` because otherwise the result of `defined_property_attributes` or `defined_property_associations` might be different before and after record save.